tools/cldr-apps/src/main/java/org/unicode/cldr/web/CookieSession.java

```diff
@@ -34,926 +33,1012 @@
-     *
-     * @param key
-     *            the key to load
-     * @return named pref, or null as default
+     * max guests allowed before guests start getting shut out (should be < CLDR_MAX_USERS)
      */
-    String prefGet(String key) {
-        String b = (String) prefs.get(key);
-        if (b == null) {
-            return null;
-        } else {
-            return b;
-        }
-    }
+    CLDR_MAX_GUESTS(0), // zero means tooManyGuests == tooManyUsers
 
     /**
-     * Store a boolean preference value
-     *
-     * @param key
-     *            the pref to put
-     * @param value
-     *            boolean value
+     * how many seconds guest can be absent/inactive before getting kicked
      */
-    void prefPut(String key, boolean value) {
-        prefs.put(key, new Boolean(value));
-    }
+    CLDR_GUEST_TIMEOUT_SECS(5 * 60), // 5 minutes
 
     /**
-     * Store a string preference value
-     *
-     * @param key
-     *            the pref to put
-     * @param value
-     *            string value
+     * how many seconds logged-in user can be absent/inactive before getting kicked
      */
-    void prefPut(String key, String value) {
-        prefs.put(key, value);
-    }
+    CLDR_USER_TIMEOUT_SECS(30 * 60); // 30 minutes
+
+    private int defVal;
+    private Integer value;
 
     /**
-     * Fetch a hashtable of per-locale session data. Will create one if it
-     * wasn't already there.
-     *
-     * @return the locale hashtable
+     * Construct a new parameter
+     * @param name
+     * @param defVal
      */
-    public Hashtable<String, Hashtable<String, Object>> getLocales() {
-        synchronized (stuff) {
-            Hashtable<String, Hashtable<String, Object>> l = (Hashtable<String, Hashtable<String, Object>>) get("locales");
-            if (l == null) {
-                l = new Hashtable<>();
-                put("locales", l);
-            }
-            return l;
-        }
+    Params(int defVal) {
+      this.defVal = defVal;
     }
 
     /**
-     * utility function for doing an encoding of a long. Creates a base26 (a-z)
-     * representation of the number, plus a leading '-' if negative. TODO:
-     * shrink? base64?
-     *
-     * @param l
-     *            some number
-     * @return string
+     * Get the param's value
+     * @return
      */
-    public static String cheapEncode(long l) {
-        StringBuilder out = new StringBuilder(10);
-        if (l < 0) {
-            out.append("-");
-            l = 0 - l;
-        } else if (l == 0) {
-            return "0";
+    public final int value() {
+      if (value == null) {
+        value =
+          CLDRConfig
+            .getInstance()
+            .getProperty(this.name().toUpperCase(), defVal);
+        SurveyLog.warnOnce(
+          logger,
+          "CookieSession: " +
+          this.name().toUpperCase() +
+          "=" +
+          value +
+          " (default: " +
+          defVal +
+          ")"
+        );
```

Comment:
this seems less clear to me. It's a log message, and could be refactored to use a messageformat. But changing it to a 10-line clause takes up more space proportionately than the debug message's impact on the code.

Reply:
function length can serve as an approximate measurement of function complexity, and when I say a function is "too long" the complexity is my main concern. In that respect, reformatting is harmless since it doesn't actually increase complexity. If it makes a function no longer fit on the screen without scrolling, then the function was probably too long to begin with.

still you have a good point that that "warnOnce" call ends up hogging a lot of space on the screen... an added incentive to use more subroutines...